### DISCUSSION TITLE
GCC 7.1 compile error.

### DISCUSSION THREAD

**Discussion by KP71156** [harfbuzz/harfbuzz#3331] [Q&A] [answered]
I'm trying to compile HarfBuzz on Linux GCC 7.1 and getting an error in hb-algs.hh:
Harfbuzz/src/hb-algs.hh:336:23: error: expected identifier before ‘&&’ token
   auto operator () (_T&& _v) const HB_AUTO_RETURN \
Code is compiling without problem on Windows.

Pulled latest code from 3.1.2 tag.
Any help would be greatly appreciated.
Ken

**Comment by behdad** [comment]
We need at least the full error log. And what Linux platform this is on, so we can try to reproduce...

**Comment by KP71156** [comment]
I'm using CentOS 7.4 with GCC 7.1 installed, here is the build log, do you need the CMake log as well?
[HarfBuzzBuildLog.txt](https://github.com/harfbuzz/harfbuzz/files/7694341/HarfBuzzBuildLog.txt)

**Reply by behdad** [reply]
I think that's supposed to work. @khaledhosny @ebraminio Don't we have a bot like that?
@matthiasclasen what's the default compiler on CentOS 7.4?

**Comment by KP71156** [comment]
I believe the default complier for CentOS 7 is GCC 4.8, but that should be irrelevant since we are overriding with GCC 7.1.

**Reply by behdad** [reply]
Okay let me see if I can somehow reproduce that on godbolt.

**Reply by behdad** [reply]
Can you attach the preprocessed output of one of the failing sources? hb-common.cc for example?

**Reply by behdad** [reply]
Either that, or ssh access to a machine I can try this...

**Reply by KP71156** [reply]
I'm not quite sure where to find the preprocessed output, or do you mean the CMake output?

**Reply by KP71156** [reply]
Sorry, can't give ssh access to internal systems.

**Comment by behdad** [comment, marked as answer]
Let's continue in https://github.com/harfbuzz/harfbuzz/issues/3332
